hikari/impl/special_endpoints.py

```diff
@@ -1005,20 +998,23 @@ def options(self) -> typing.Sequence[commands.CommandOption]:
 
     def build(self, entity_factory: entity_factory_.EntityFactory, /) -> data_binding.JSONObjectBuilder:
         data = super().build(entity_factory)
-        data["type"] = commands.CommandType.SLASH
         data.put("description", self._description)
         data.put_array("options", self._options, conversion=entity_factory.serialize_command_option)
         return data
 
 
+@attr_extensions.with_copy
 @attr.define(kw_only=False, weakref_slot=False)
 class ContextMenuCommandBuilder(CommandBuilder, special_endpoints.ContextMenuCommandBuilder):
     """Builder class for context menu commands."""
 
-    def build(self, entity_factory: entity_factory_.EntityFactory, /) -> data_binding.JSONObjectBuilder:
-        data = super().build(entity_factory)
-        data["type"] = self._type
-        return data
+    _type: commands.CommandType = attr.field()
+    # name is re-declared here to ensure type is before it in the initializer's args.
+    _name: str = attr.field()
```

Comment:
```suggestion
    _type: commands.CommandType = attr.field(kw_only=True)
    # name is re-declared here to ensure type is before it in the initializer's args.
    _name: str = attr.field(kw_only=True)
```

This is a bit from memory, since github mobile doesnt let me investigate nicely, but I believe the others were marked as kwargs only

Reply:
These weren't since they're required, this just matches the pattern found in other parts of the library for signatures